admin/listPurchases/listView.html

```diff
@@ -295,25 +295,29 @@
     });
 }
 
-function requestOrders(callback) {
-    var req = new XMLHttpRequest();
-    req.open("GET", "/admin/orders");
-    req.setRequestHeader("Content-Type", "application/json");
-    req.onreadystatechange = function() {
-        if(req.readyState != 4) {
-            return;
-        }
-        if(req.status == 200) {
-            return callback(null, JSON.parse(this.responseText));
+async function requestOrders(callback) {
+    let response;
+    try{
+        response = await fetch("/admin/orders", {
+            method: 'GET',
+            headers: {
+                "Content-Type": "application/json",
+            },
+        });
+        if(!response.ok){
+            callback({
+                trace: new Error("Error response from server on request"),
+                status: response.status
```

Comment:
It would be nice to also get the response data out here. If we cannot be sure that it is JSON, at least try to get it as text, and handle the case where that fails by setting it to undefined or something.

What the response says is often significant to what kind of error is going on.